exonum/src/blockchain/mod.rs

```diff
@@ -275,6 +272,28 @@ impl Blockchain {
                 schema.tx_location_by_tx_hash_mut().put(hash, location);
             }
 
+            // Invoke handle_execute for each service
+            for service in self.service_map.values() {
+                fork.checkpoint();
+                match panic::catch_unwind(panic::AssertUnwindSafe(
```

Comment:
Nitpick: This formatting looks a bit uncomfortable to me; I'd propose to use a variable for the `catch_unwind` result here.